src/Raven.Server/Documents/Indexes/IndexStateRecord.cs

```diff
@@ -89,6 +92,23 @@ public void Recreate(Func<IndexSearcher> createIndexSearcher)
     {
         if (IndexSearcher.IsValueCreated == false)
             return;
+
+        var old = IndexSearcher.Value;
+
+        InstancesThatGotRecreated.Add(old, new IndexSearcherDisposer(old)); // this way we ensure the old searcher will get disposed - https://www.ayende.com/blog/199169-A/externalfinalizer-adding-a-finalizer-to-3rd-party-objects
+
         IndexSearcher = new Lazy<IndexSearcher>(createIndexSearcher);
     }
+
+    private class IndexSearcherDisposer(IndexSearcher searcher)
+    {
+        ~IndexSearcherDisposer()
+        {
+            using (searcher)
+            using (searcher.IndexReader)
+            {
+                
+            }
```

Comment:
Do we need to have `try/catch` here?

Reply:
See: https://github.com/ravendb/ravendb/blob/cf0110d66970e4899f868d5ea6cc0dcb0848dadb/src/Raven.Server/Documents/Indexes/IndexStateRecord.cs#L73